src/home/homework-mount.c

```diff
@@ -61,7 +62,14 @@ int home_mount_node(
                         return log_oom();
         }
 
-        if (!strextend_with_separator(&joined, ",", discard ? "discard" : "nodiscard"))
+        const char *discard_mount_option = "discard";
+        if (streq(fstype, "btrfs")) {
+                Condition *c = condition_new(CONDITION_KERNEL_VERSION, ">= 5.6", false, false);
```

Comment:
we frown on kernel version checks, since many distros (in particular commercial ones) are really keen on backporting features. That means version checks are not useful for checking feature availability.

Can we explicitly check if this feature is available already? i.e. btrfs has `/sys/fs/btrfs/features/`. Can we check availbility there? If not, is there maybe a closely related feature listed there that came at the same time, and unlikely to be skipped if async discard is backported?

What is the actual failure mode of the new discard option on old kernels? is it just ignored? or does the whole mount() syscall fail? If the latter, we could just try and fall back.